api/suggested-fees.ts

```diff
@@ -78,6 +85,46 @@ const handler = async (
     ]);
     const { l1Token, hubPool, chainId: computedOriginChainId } = tokenDetails;
 
+    if (sdk.utils.isDefined(message) && !sdk.utils.isMessageEmpty(message)) {
+      if (message.length % 2 !== 0) {
+        // Our message encoding is a hex string, so we need to check that the length is even.
+        throw new InputError("Message must be an even hex string");
+      }
+      const isRecipientAContract = await sdk.utils.isContractDeployedToAddress(
+        recipientAddress,
+        provider
+      );
+      if (!isRecipientAContract) {
+        throw new InputError(
+          "Recipient address with a contract deployed to it is required when a message is provided"
```

Comment:
```suggestion
          "Recipient must be a contract when a message is provided"
```

This is needlessly complicated IMO